###### Assessment

###### I am going to provide two .csv files , you are supposed to work on them and have to provide solutions to the following problems

###### import necessary libraries

In [2]:
import pandas as pd
import numpy as np

###### merge those two csv files (after getting as dataframes, get them as a single dataframe)

In [3]:
df_1=pd.read_csv('/content/college_1.csv')
df_2=pd.read_csv('/content/college_2.csv')
df_1

Name  python  mysql  Previous Geekions  CodeKata Score  \
0        A.Dharani    82.0   20.0              24500           24500   
1      V.JEEVITHA     82.0   20.0              21740           21740   
2      HEMAVATHI.R   100.0  100.0              19680           19680   
3      Mugunthan S   100.0   47.0              10610           10610   
4      Sathammai.S   100.0    8.0               8980            8980   
..             ...     ...    ...                ...             ...   
79    KarthikeyanS    45.0    0.0                  0               0   
80        BARATH.P    29.0    0.0                  0               0   
81   N.Ajith kumar    82.0    0.0                  0               0   
82    mohamed nabi     0.0    0.0                  0               0   
83  yaser ahamed.A     0.0   27.0                  0               0   

                                Department  Rising  
0         Computer Science and Engineering       0  
1         Computer Science and Engineering       0  
2         Computer Science and Engineering       0  
3         Computer Science and Engineering       0  
4         Computer Science and Engineering       0  
..                                     ...     ...  
79  Electronics and Electrical Engineering       0  
80  Electronics and Electrical Engineering       0  
81  Electronics and Electrical Engineering       0  
82  Electronics and Electrical Engineering       0  
83  Electronics and Electrical Engineering       0  

[84 rows x 7 columns]

In [4]:
df_2

Name  python  python_en  mysql  computational_thinking  \
0            ASHOK KUMAR K     100          0     31                       9   
1                  Chandru       0         20      0                       0   
2        Ganesh Ramkumar R      -1         55     24                       6   
3          Bodipudi Harini      -1          0     35                      -1   
4                VIINU V P      -1        100     24                      -1   
5               Poojitha Y       0         78     35                       0   
6   Gowtham Chandrasekaran      -1          0      0                       3   
7                Gowtham R      -1         60      4                       6   
8                 shifak N      58          0      0                       0   
9                  Balaji       -1        100     20                       0   
10                Rushitha      -1         15     62                       9   
11    Balaraman Muthupandi      -1          9      0                       0   
12                JPradhap     100          0      0                       0   
13           SRIRAMKUMAR B      -1          0      0                       0   
14               Praneetha      -1         23    100                      -1   
15             Rahul Raj K      -1          0      0                       0   
16                 Guhan S      -1         72     35                      39   
17            Suryarajan S      -1         63     31                       0   
18          Narasimhan Y L      -1         -1      0                       0   
19            Leelakrishna      -1          6      4                       0   
20          PremsagarReddy      -1         20      0                       0   
21               pradeeban      16         20     24                       0   
22             santhoshini      -1         43      0                       0   
23              MUKESH S M      -1         46      0                       0   
24           P.GOVARDHANAN      -1         40      0                      -1   
25            SHAIK DANISH      -1         43      0                       0   
26               sridhar.s      -1         52      0                       0   
27       Vimalesh Fernando      -1          0      0                       0   
28           M.Karthikeyan      -1          0      4                       6   
29               Abishak s      -1         55     35                       6   
30           praveen raj j      24         -1      0                       0   
31              AMARNATH D      -1         52     12                      -1   
32                    bala      32         49      0                      -1   
33                    XY Z      -1         20     -1                      -1   
34               Hariharan      -1         -1     -1                       0   

    Previous Geekions  CodeKata Score  \
0               19400           19400   
1               14150           14490   
2                8790           10790   
3               10040           10040   
4                9150            9150   
5                7210            8970   
6                7250            8950   
7                8650            8650   
8                5180            8320   
9                6170            8160   
10               7470            8090   
11               7670            8050   
12               7340            8030   
13               6400            7940   
14               7170            7730   
15               6280            7640   
16               7310            7630   
17               6710            7550   
18               4800            6800   
19               5300            6640   
20               5290            6290   
21               6060            6090   
22               3910            5740   
23               5200            5200   
24               5050            5050   
25               4050            4820   
26               3860            4

In [5]:
df_3=pd.concat([df_1,df_2])

The merged DataFrame is as follows

In [6]:
df_3

Name  python  mysql  Previous Geekions  CodeKata Score  \
0       A.Dharani    82.0   20.0              24500           24500   
1     V.JEEVITHA     82.0   20.0              21740           21740   
2     HEMAVATHI.R   100.0  100.0              19680           19680   
3     Mugunthan S   100.0   47.0              10610           10610   
4     Sathammai.S   100.0    8.0               8980            8980   
..            ...     ...    ...                ...             ...   
30  praveen raj j    24.0    0.0               2380            2380   
31     AMARNATH D    -1.0   12.0               1890            1890   
32           bala    32.0    0.0               1720            1720   
33           XY Z    -1.0   -1.0                  0               0   
34      Hariharan    -1.0   -1.0                  0               0   

                                   Department  Rising  python_en  \
0            Computer Science and Engineering       0        NaN   
1            Computer Science and Engineering       0        NaN   
2            Computer Science and Engineering       0        NaN   
3            Computer Science and Engineering       0        NaN   
4            Computer Science and Engineering       0        NaN   
..                                        ...     ...        ...   
30           Computer Science and Engineering       0       -1.0   
31  Electronics and Communication Engineering       0       52.0   
32  Electronics and Communication Engineering       0       49.0   
33           Computer Science and Engineering       0       20.0   
34           Computer Science and Engineering       0       -1.0   

    computational_thinking  
0                      NaN  
1                      NaN  
2                      NaN  
3                      NaN  
4                      NaN  
..                     ...  
30                     0.0  
31                    -1.0  
32                    -1.0  
33                    -1.0  
34                     0.0  

[119 rows x 9 columns]

###### Take each csv file , split that csv file into multiple categories (example csv files are added in the repo) 


###### consider if the codekata score exceeds 15000 points(present week) then make a csv on those observations as Exceeded expectations.csv


###### if  10000<codekata score<15000   (Reached_expectations.csv)



###### if  7000<codekata score<10000   (Needs_Improvement.csv)


###### if  codekate score < 7000        (Unsatisfactory.csv)

In [14]:
Exceeded_expectations=df_3[(df_3['CodeKata Score']>15000)]
Exceeded_expectations

Name  python  mysql  Previous Geekions  CodeKata Score  \
0      A.Dharani    82.0   20.0              24500           24500   
1    V.JEEVITHA     82.0   20.0              21740           21740   
2    HEMAVATHI.R   100.0  100.0              19680           19680   
0  ASHOK KUMAR K   100.0   31.0              19400           19400   

                               Department  Rising  python_en  \
0        Computer Science and Engineering       0        NaN   
1        Computer Science and Engineering       0        NaN   
2        Computer Science and Engineering       0        NaN   
0  Electronics and Electrical Engineering       0        0.0   

   computational_thinking  
0                     NaN  
1                     NaN  
2                     NaN  
0                     9.0

In [15]:
Exceeded_expectations.to_csv('Exceeded expectations.csv')

In [16]:
Reached_expectations=df_3[(df_3['CodeKata Score']>10000)&(df_3['CodeKata Score']<15000)]
Reached_expectations

Name  python  mysql  Previous Geekions  CodeKata Score  \
3        Mugunthan S   100.0   47.0              10610           10610   
1            Chandru     0.0    0.0              14150           14490   
2  Ganesh Ramkumar R    -1.0   24.0               8790           10790   
3    Bodipudi Harini    -1.0   35.0              10040           10040   

                                  Department  Rising  python_en  \
3           Computer Science and Engineering       0        NaN   
1           Computer Science and Engineering     340       20.0   
2           Computer Science and Engineering    2000       55.0   
3  Electronics and Communication Engineering       0        0.0   

   computational_thinking  
3                     NaN  
1                     0.0  
2                     6.0  
3                    -1.0

In [17]:
Reached_expectations.to_csv('Reached_expectations.csv')

In [18]:
Needs_improvement=df_3[(df_3['CodeKata Score']>7000)&(df_3['CodeKata Score']<10000)]
Needs_improvement

Name  python  mysql  Previous Geekions  CodeKata Score  \
4              Sathammai.S   100.0    8.0               8980            8980   
5             NIVEESHWAR S   100.0    0.0               8200            8200   
6            KALAIARASAN K    85.0   47.0               7750            7880   
7     MOHAMED ZUBAIR AHMED    82.0    4.0               7020            7020   
4                VIINU V P    -1.0   24.0               9150            9150   
5               Poojitha Y     0.0   35.0               7210            8970   
6   Gowtham Chandrasekaran    -1.0    0.0               7250            8950   
7                Gowtham R    -1.0    4.0               8650            8650   
8                 shifak N    58.0    0.0               5180            8320   
9                  Balaji     -1.0   20.0               6170            8160   
10                Rushitha    -1.0   62.0               7470            8090   
11    Balaraman Muthupandi    -1.0    0.0               7670            8050   
12                JPradhap   100.0    0.0               7340            8030   
13           SRIRAMKUMAR B    -1.0    0.0               6400            7940   
14               Praneetha    -1.0  100.0               7170            7730   
15             Rahul Raj K    -1.0    0.0               6280            7640   
16                 Guhan S    -1.0   35.0               7310            7630   
17            Suryarajan S    -1.0   31.0               6710            7550   

                                   Department  Rising  python_en  \
4            Computer Science and Engineering       0        NaN   
5            Computer Science and Engineering       0        NaN   
6   Electronics and Communication Engineering     130        NaN   
7            Computer Science and Engineering       0        NaN   
4      Electronics and Electrical Engineering       0      100.0   
5            Computer Science and Engineering    1760       78.0   
6            Computer Science and Engineering    1700        0.0   
7   Electronics and Communication Engineering       0       60.0   
8      Electronics and Electrical Engineering    3140        0.0   
9      Electronics and Electrical Engineering    1990      100.0   
10           Computer Science and Engineering     620       15.0   
11           Computer Science and Engineering     380        9.0   
12     Electronics and Electrical Engineering     690        0.0   
13           Computer Science and Engineering    1540        0.0   
14  Electronics and Communication Engineering     560       23.0   
15           Computer Science and Engineering    1360        0.0   
16  Electronics and Communication Engineering     320       72.0   
17           Computer Science and Engineering     840       63.0   

    computational_thinking  
4                      NaN  
5                      NaN  
6                      NaN  
7                      NaN  
4                     -1.0  
5                      0.0  
6                      3.0  
7                      6.0  
8                      0.0  
9                      0.0  
10                     9.0  
11                     0.0  
12                     0.0  
13                     0.0  
14                    -1.0  
15                     0.0  
16                    39.0  
17                     0.0

In [19]:
Needs_improvement.to_csv('Needs_improvement.csv')

In [20]:
Unsatisfactory=df_3[(df_3['CodeKata Score']<7000)]
Unsatisfactory

Name  python  mysql  Previous Geekions  CodeKata Score  \
8        J.SUGANTHI    27.0  50.00               5860            5860   
9      thamizhpaana    29.0   0.00               5500            5500   
10  Iyappan Samiraj    50.0  24.00               5400            5400   
11     Ponniyamma.R   100.0  11.25               5380            5380   
12        SABAPATHI   100.0   0.00               5280            5280   
..              ...     ...    ...                ...             ...   
30    praveen raj j    24.0   0.00               2380            2380   
31       AMARNATH D    -1.0  12.00               1890            1890   
32             bala    32.0   0.00               1720            1720   
33             XY Z    -1.0  -1.00                  0               0   
34        Hariharan    -1.0  -1.00                  0               0   

                                   Department  Rising  python_en  \
8   Electronics and Communication Engineering       0        NaN   
9            Computer Science and Engineering       0        NaN   
10  Electronics and Communication Engineering       0        NaN   
11           Computer Science and Engineering       0        NaN   
12           Computer Science and Engineering       0        NaN   
..                                        ...     ...        ...   
30           Computer Science and Engineering       0       -1.0   
31  Electronics and Communication Engineering       0       52.0   
32  Electronics and Communication Engineering       0       49.0   
33           Computer Science and Engineering       0       20.0   
34           Computer Science and Engineering       0       -1.0   

    computational_thinking  
8                      NaN  
9                      NaN  
10                     NaN  
11                     NaN  
12                     NaN  
..                     ...  
30                     0.0  
31                    -1.0  
32                    -1.0  
33                    -1.0  
34                     0.0  

[93 rows x 9 columns]

In [21]:
Unsatisfactory.to_csv('Unsatisfactory.csv')

###### Average of previous week geekions vs this week geekions (i.e Previous Geekions vs CodeKata Score)

###### No of students participated 

In [37]:
df_3['Name'].unique()

array(['A.Dharani', 'V.JEEVITHA ', 'HEMAVATHI.R', 'Mugunthan S',
       'Sathammai.S', 'NIVEESHWAR S', 'KALAIARASAN K',
       'MOHAMED ZUBAIR AHMED', 'J.SUGANTHI', 'thamizhpaana',
       'Iyappan Samiraj', 'Ponniyamma.R', 'SABAPATHI', 'RAGHAVAN',
       'Melvin Dukles.R', 'Pavithra', 'PRIYANKA .S', 'v.vithyasri',
       'Abirami Anbazhagan', 'S.srinivasan', 's.janani', 'kamalishwari',
       'Yuvasri', 'balaji', 'MEERA.M', 'Tpavithra', 'sri dharshini',
       'shankari', 'pooja priya.B', 'chithra', 'manikandan',
       'Kaliaperumal G', 'SarumathiR', 'SOWMIYA.V', 'tamilarasi',
       'Barath D', 'selvakumar', 'nivedha.s', 'Stephen', 'vignesh',
       'HSharukhan', 'vijayakumar', 'ABIRAMI.S', 'v.gowsalya', 'ponni.p',
       'PRAVEENA K', 'siva sai perumal polisetti', 'MadhumathiB',
       'b.akshaya', 'bavithra', 'Deivani.S', 'Nivetha.S', 'GUNAMATHI.S',
       'RANJITH.R', 'u.abarna', 'kaviyarasi.k', 'pravina', 'M.PRIYA',
       'M.Nisha@mary', 'Thivviya', 'muvendiran.m', 'Aravind', 'H

In [38]:
df_3.shape

(119, 9)

Totally 119 students participated

###### #Average completion of python course or my_sql or python english or computational thinking

In [46]:
np.average(df_3['python'])

54.35294117647059

In [47]:
np.average(df_3['mysql'])

23.5

In [50]:
np.average(df_2['python_en'])

29.65714285714286

In [51]:
np.average(df_2['computational_thinking'])

2.2

###### rising star of the week (top 3 candidate who performed well in that particular week)

In [22]:
Exceeded_expectaions['Name'][0:3]

0      A.Dharani
1    V.JEEVITHA 
2    HEMAVATHI.R
Name: Name, dtype: object

###### Shining stars of the week (top 3 candidates who has highest geekions)

In [23]:
Exceeded_expectaions['Name'][0:3]

0      A.Dharani
1    V.JEEVITHA 
2    HEMAVATHI.R
Name: Name, dtype: object

###### Department wise codekata performence (pie chart)

In [24]:
import plotly.express as px
fig=px.pie(df_3,values='CodeKata Score',names='Department',title='Department wise codekata performence')
fig.show()

###### Department wise toppers (horizantal bar graph or any visual representations of your choice)

In [53]:
import plotly.express as px
fig=px.bar(df_3,x='Department',y='CodeKata Score',color='Name')
fig.show()

Based on the above bar graph we can say that

praveen raj j of Computer Science and Engineering department has got top
CodeKata score

bala of Electronics adn Communication Enginneering department has got top CodeKata score

J Pradhap of Electronics and Electrical Engineering department has got top
CodeKata score